compute_endpoint/globus_compute_endpoint/engines/base.py

```diff
@@ -117,6 +119,55 @@ def _status_report(self, shutdown_event: threading.Event, heartbeat_period: floa
             packed = messagepack.pack(status_report)
             self.results_passthrough.put({"message": packed})
 
+    def _handle_task_exception(
+        self,
+        task_id: str,
+        execution_begin: TaskTransition,
+        exception: BaseException,
+    ) -> bytes:
+        """Repackage task exception to messagepack'ed bytes
+        Parameters
+        ----------
+        task_id: str
+        execution_begin: TaskTransition
+        exception: Exception object from the task failure
+
+        Returns
+        -------
+        bytes
+        """
+        code, user_message = get_result_error_details(exception)
+        error_details = {"code": code, "user_message": user_message}
+        execution_end = TaskTransition(
+            timestamp=time.time_ns(),
+            actor=ActorName.INTERCHANGE,
+            state=TaskState.EXEC_END,
+        )
+        exception_string = ""
+        for index, prev_exc in enumerate(
+            self._retry_table[task_id]["exception_history"]
+        ):
+            templated_history = f"""
+++++++++++++++++++++ Traceback from attempt:{index} ++++++++++++++++++++
```

Comment:
I think `index + 1` would be more clear in this instance.

Reply:
I've sort of cleaned this up.